app/controllers/sp_components_controller.rb

```diff
@@ -13,6 +13,7 @@ def new
 
   def show
     @component = SpComponent.find(params[:id])
+    @available_services = Service.where(sp_component_id: [nil, ''])
```

Comment:
Apologies if I'm missing the context here, but this will just return an empty relation so the block in the view will never be shown

Reply:
avialable_services are Service entries without an associated **sp_component_id**

Reply:
this would be a great candidate for scopes